service/openapi.yaml

```diff
@@ -0,0 +1,87 @@
+openapi: "3.0.0"
+info:
+  version: 0.1.0
+  title: OpenFeature Remote Evaluation Protocol (OFREP)
+  description: OFREP define the protocol for remote flag evaluations
+  contact:
+    url: https://github.com/open-feature/protocol
+  license:
+    name: Apache 2.0
+paths:
+  /evaluate:
+    post:
+      description: OFREP flag evaluation request
+      requestBody:
+        content:
+          'application/json':
+            schema:
+              $ref: '#/components/schemas/evaluationRequest'
+      responses:
+        '200':
+          description: OFREP successful evaluation response
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/success"
+        '400':
+          description: OFREP error response
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/failure"
+components:
+  schemas:
+    evaluationRequest:
+      description: Flag evaluation request
+      properties:
+        flagKey:
+          $ref: "#/components/schemas/flagKey"
+        flagType:
+          type: string
+          enum: [string, boolean, integer, float, object]
+          description: Type to evaluate the given flag
+        context:
+          type: object
+          description: Context information for flag evaluation
+      required:
+        - flagKey
+        - flagType
+    success:
+      description: Flag evaluation success response.
+      properties:
+        flagKey:
+          $ref: "#/components/schemas/flagKey"
+        value:
```

Comment:
I believe the type should be included in the response.

Reply:
Update: I don't think the type is necessary in the request or response. Developers must specify the expected type when evaluating a feature flag. The provider can then perform a runtime type check. I've opened up a separate PR containing the proposed changes.

Reply:
As I said in the beginning the problem is more with double vs int because JSON does not make the difference and this is where it becomes tricky.

For example, in GO all integers are marshall in float which makes it complex to handle properly the TYPE_MISMATCH between those 2 types, we are kind of lost in the JSON translation.
If we request to be explicit on the type from the flag management system, we will reduce the choice that the provider should take.
I think that having the type in the response is a good approach here and it will simplify a lot how to build providers.

The example I have in mind is, that you can receive a int that looks like:
```json
...
"value":1.0
...
```

How do you know if it is an `int` or a `float`? We are missing context in the provider to know if we should return a `TYPE_MISMATCH` error or if we should cast the value into an `int`.

Reply:
To add on to this. Even if, at one point in time, you get a value that looks like `"value":1`, then treating that as specifically an integer can still cause you trouble.

The consumer of the code handles it as an integer only, and then someone changes the value to `0.5` and now the type becomes a double and things break.

"I want this thing, and I want it as an int" is a safer interface.

It doesn't happen as often, but this is also true for other types. If a flag was a string, and it gets changes to an int, then the interface of "get me this thing as a string" can still provide a correct default value and indicate a `TYPE_MISMATCH`, and it can be done at the provider level, which means vendor analytics are still correct.

Reply:
The tricky part of asking in the API request will be that we are not always requesting a flag, in the bulk evaluation we can ask for everything available.

Reply:
If you don't want to transmit it in the request, and you just want to get whatever type it is back, then likely you need another bit of communication. Something that lets the provider know the outcome. The type wasn't what was expected and we used the default value. (Though this is a harder sell because now the vendor SDK needs a way to ingest that information as well.)

As it currently is I think the default value information is missing as well? I have concerns that the fidelity of information getting back to the provider is going to hinder things.

Reply:
The idea is to send back the type configured from the flag in the response, so it is explicit and in the provider you can compare the method called and the type configured in flag management system.

Let's say that you configure a flag as double in LD, the bulk API response will be something like this:
```json
[
  {
    "type": "double",
    "value": 1,
    "key": "my-flag",
    "reason": "TARGETING_MATCH",
    "variant": "A",
  },
  {
    "type": "string",
    "value": "my-flag-value",
    "key": "my-flag",
    "reason": "TARGETING_MATCH",
    "variant": "B"
  }
]
```

With the `type` you know what is expected, allowing you to cast the `1` into a double with no doubt because you know the type of the flag.

It means that if you call `client.getIntegerValue("my-flag", 5.1);` you should expect an errorCode `TYPE_MISMATCH` even if in the JSON format it looks like an integer.

Reply:
If I was using this I would want to know that using the remote evaluation is equivalent to in-process evaluation in terms of results and ideally analytics. I don't think that will be true with this approach, it reverses the driver of typing.

An individual evaluation call through remote evaluation may produce different results, have different constraints, and different produce different analytics.

For example the LD case would encounter at least these:
1.) Only double support for evaluation methods.
2.) The provider wouldn't have a way to produce wrong type analytics.
3.) The provider wouldn't have a way to provide default values in analytics.

So if you use remote eval you are basically getting a different product, not just out of process eval.

Reply:
In the community meeting on 29.02.2024, we decided not to include any type details in either request and response

This decision is now documented in a dedicated ADR [1]. We could add these details if we see the requirement in a future iteration. 

If no objections, I will resolve this conversation before merging the PR. 

[1] - https://github.com/open-feature/protocol/blob/0896456b001f217c53a155d6075854ef58ff06b2/service/adrs/0003-no-flag-type-in-request-or-response.md